modules/custom/moj_resources/src/Utilities.php

```diff
@@ -34,4 +38,88 @@ public static function filterByPrisonCategories($prisonId, $prisonCategories, $q
       ->condition($filterByPrisonCategories)
       ->condition($filterByPrisonId);
   }
+
+  /**
+   * Loads term for ID
+   *
+   * @param int $termId
+   * @param EntityTypeManagerInterface $termStorage
+   *
+   * @return EntityInterface
+  */
+  public static function getTermFor($termId, $termStorage) {
+    $term = $termStorage->load($termId);
+
+    if (!$term) {
+      throw new NotFoundHttpException(
+        'Term not found',
+        null,
+        404
+      );
+    }
+
+    return $term;
+  }
+
+  /**
+   * Loads node for ID
+   *
+   * @param int $nodeId
+   * @param EntityTypeManagerInterface $nodeStorage
+   *
+   * @return EntityInterface
+  */
+  public static function getNodeFor($nodeId, $nodeStorage) {
+    $node = $nodeStorage->load($nodeId);
+
+    if (!$node) {
+      throw new NotFoundHttpException(
+        'Node not found',
+        null,
+        404
+      );
+    }
+
+    return $node;
+  }
+
+  /**
+   * Get Prisons for a Drupal node object
+   *
+   * @param EntityInterface $node
+   * @return int[]
+  */
+  public static function getPrisonsFor($node) {
+    $prisons = array();
+
+    foreach ($node->field_moj_prisons as $prison) {
+      array_push($prisons, $prison->target_id);
+    }
+
+    return $prisons;
+  }
+
+  /**
+   * Get Prison Categories for a Drupal node object
+   *
+   * @param EntityInterface $term
+   * @return int[]
+  */
+  public static function getPrisonCategoriesFor($node) {
+    $prisonCategories = array();
```

Comment:
Amusingly, `$a = [];` is not the same as `$a = array();`. The former causes `count()` to return `1` due to 

> When the parameter is neither an array nor an object with implemented Countable interface, 1 will be returned

Reply:
Interesting, I'm assuming this is similar to in JS with `[1, 2, 3]` and `new Array(1, 2, 3)`?

Reply:
It caused me a lot of swearing when I could see the array was empty when printed out but still didn't pass the count.

Reply:
I've also noticed similar issues with `Entity` objects returned by Drupal, they're iterable, but don't work with all Array functions